fabric/main.py

```diff
@@ -125,6 +126,7 @@ def is_classic_task(tup):
             callable(func)
             and (func not in _internals)
             and not name.startswith('_')
+            and not (inspect.isclass(func) and issubclass(func, Exception))
```

Comment:
Class-based tasks are actually a definite (if minority) use case, so the limitation of "it's an Exception" should stay, IMHO.